client/src/components/MultiJurisdictionAudit/RoundManagement/useBatchResults.ts

```diff
@@ -0,0 +1,124 @@
+import { useEffect, useState } from 'react'
+import { toast } from 'react-toastify'
+import { api } from '../../utilities'
+
+export interface IResultValues {
+  [batchId: string]: {
+    [choiceId: string]: number | string
+  }
+}
+
+export interface IBatch {
+  id: string
+  name: string
+  numBallots: number
+  auditBoard: null | {
+    id: string
+    name: string
+  }
+}
+
+const stringifyPossibleNull = (v: string | number | null) => (v ? `${v}` : '')
+
+const reformatResults = (r: IResultValues, numberify = true): IResultValues => {
+  return Object.keys(r).reduce(
+    (a, batchId) => ({
+      ...a,
+      [batchId]: Object.keys(r[batchId]).reduce(
+        (b, choiceId) => ({
+          ...b,
+          [choiceId]: numberify
+            ? Number(r[batchId][choiceId])
+            : stringifyPossibleNull(r[batchId][choiceId]),
+        }),
+        {}
+      ),
+    }),
+    {}
+  )
+}
+
+const numberifyResults = (r: IResultValues): IResultValues => reformatResults(r)
+
+const getResults = async (
+  electionId: string,
+  jurisdictionId: string,
+  roundId: string
+): Promise<IResultValues | null> => {
+  try {
+    const response: IResultValues = await api(
+      `/election/${electionId}/jurisdiction/${jurisdictionId}/round/${roundId}/batches/results`
+    )
+    return reformatResults(response, false)
+  } catch (err) /* istanbul ignore next */ {
+    // TODO move toasting into api
+    toast.error(err.message)
+    return null
+  }
+}
+
+const getBatches = async (
+  electionId: string,
+  jurisdictionId: string,
+  roundId: string
+): Promise<IBatch[] | null> => {
+  try {
+    const { batches }: { batches: IBatch[] } = await api(
+      `/election/${electionId}/jurisdiction/${jurisdictionId}/round/${roundId}/batches`
+    )
+    return batches
+  } catch (err) /* istanbul ignore next */ {
+    // TODO move toasting into api
+    toast.error(err.message)
+    return null
+  }
+}
+
+const useBatchResults = (
+  electionId: string,
+  jurisdictionId: string,
+  roundId: string
+): [
+  IResultValues | null,
+  IBatch[] | null,
+  (arg0: IResultValues) => Promise<boolean>
+] => {
+  const [results, setResults] = useState<IResultValues | null>(null)
+  const [batches, setBatches] = useState<IBatch[] | null>(null)
+
+  const updateResults = async (newResults: IResultValues): Promise<boolean> => {
+    /* istanbul ignore next */
+    if (!results) return false
```

Comment:
not sure what this line is for

Reply:
@jonahkagan Oh dear, haha. I keep meaning to ask you what it's there for as well. I've taken it out and put it back in and I've seen zero functionality change, but I've just figured I was missing something. :stuck_out_tongue_closed_eyes:

Reply:
uhhh i have no idea. reading it, it just seems to disable updating before loading. i don't remember writing it (did i?)

Reply:
I thought you did, but now I'm not sure, haha. I'll take it out, though.